### ISSUE TITLE
Separate log file for each test in a test-suite.

### ISSUE THREAD

**Issue by radzimir** [telekom/testerra#205]
If tests are running in parallel under one TestSuite, log file becames a chaotic mix of messages.

It would be nice to have separate log file for every Test-Class running in separate thread.

Thouse files should be then collected by ReportNG plugin.

**Comment by mreiche** [comment]
This is Log4j magic by using Appenders and maybe custom policies and not done by Testerra.

Maybe this helps
https://stackoverflow.com/questions/44287020/log-separate-log-levels-to-separate-files-in-log4j2-properties-file

But I would suggest to use other log file analyzing tools to get this done instead of splitting these files up.

**Comment by radzimir** [comment]
In ReportNGs Report under Logs all lines are prefixed with Thread name, so Testerra knows somehow which line comes from which test, but there is no filtering or download functionality there. 
Build Log from Team City do not contains any thread names, so I can't grep that either.

**Comment by mreiche** [comment]
You just need to follow the [standard logging](https://docs.testerra.io/testerra/stable/index.html#_logging).
```xml
<PatternLayout pattern="%d{dd.MM.yyyy HH:mm:ss.SSS} [%t][%p]%contextIds: %c{2} - %m{nolookups}%n" />
```

**Comment by radzimir** [comment]
I'v managed to configure separate log files using [RoutingAppender](https://logging.apache.org/log4j/2.x/manual/appenders.html#RoutingAppender). 

Where do i put thet so that they will be grabbed by Report-NG and found somhere under collected artifacts?

**Comment by mreiche** [comment]
ReportNG doesn't read log files, it has its own format in `report-ng/model/execution` which cannot be modified.
When you want to put these log files to the artifacts directory, use an additional `FileAppender`.

**Comment by radzimir** [comment]
Thanks to Testerra's plugins system, I've written SaveLogsHook pushing log files to artifacts directory. It works now. I close this request.
